src/search/landmarks/landmark_count_heuristic.cc

```diff
@@ -108,6 +112,45 @@ LandmarkCountHeuristic::LandmarkCountHeuristic(const options::Options &opts)
     }
 }
 
+int LandmarkCountHeuristic::get_min_cost_of_achievers(const set<int> &achievers,
+                                                      const TaskProxy &task_proxy) {
+    int min_cost = numeric_limits<int>::max();
+    for (int operator_or_axiom_id : achievers) {
+        min_cost = min(min_cost, get_operator_or_axiom(
+                           task_proxy, operator_or_axiom_id).get_cost());
+    }
+    return min_cost;
+}
+
+void LandmarkCountHeuristic::compute_landmark_costs() {
+    /*
+       This function runs under the assumption that landmark node IDs go
+       from 0 to the number of landmarks - 1, therefore the entry in
+       *min_first_achiever_costs* and *min_possible_achiever_costs*
+       at index i corresponds to the entry for the landmark node with ID i.
+    */
+    min_first_achiever_costs.reserve(lgraph->get_num_landmarks());
+    min_possible_achiever_costs.reserve(lgraph->get_num_landmarks());
+    for (auto &node : lgraph->get_nodes()) {
+        /*
+           Currently derived landmarks are unsupported in the admissible
+           case and ignored in the inadmissible case. We ensure the latter
+           by setting minimum achiever costs to 0.
```

Comment:
The admissible code cannot handle derived landmarks; it exits with SEARCH_UNSUPPORTED if axioms are detected. But it is admittedly not obvious that this option is only for the inadmissible setting. I suggest to add a short comment to the option documentation, e.g. "the cost to use for derived landmarks (only applicable when admissible=false, since the admissible lmcount heuristic does not support axioms)".

Reply:
I don't understand the connection between my comment and your reaction. I believe this function is only executed in the inadmissible case (maybe add an assertiaon?). Hence, it doesn't really matter here, that the admissible case cannot deal with axioms (for now).
My comment aimed at the recent change that derived landmarks aren't simply ignored anymore, but one can use the option `derived_lm_cost` to decide whether derived landmarks should be considered or not. More specifically, in lines 141--142 you set the achiever costs to this value and not always to 0, so there's an inconsistency between the code and the comment above, isn't there?